### PR TITLE
Feature/#94 motor group drivetrain

### PR DESCRIPTION
**Pull request by FRCTeam172**

### CODE REVIEW

**Review by cml1010101 — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/variants/MotorGroupTalon.java [hunk 1]
```diff
@@ -55,7 +57,7 @@ public double getEncoderRotations() {
     }
```
Comment (cml1010101) [inline src/main/java/frc/robot/subsystems/variants/MotorGroupTalon.java:57]: you need to invert the encoder rotations if necessary (if inverted)

**Review by cml1010101 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -6,76 +6,127 @@
...
+    };
+  }
   /**
    * Gets the distance traveled by the left encoder
    * @return left encoder distance in meters
    */
-  public abstract double getLeftDistance();
+  public double getLeftDistance()
+  {
+    return 0;
```
Comment (cml1010101) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:92]: should return the distance traveled. This is vital.

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -6,76 +6,127 @@
...
+  }
   /**
    * Gets the speed that the wheels are moving at
    * @return DifferentialDriveWheelSpeeds in m/s
    */
-  public abstract DifferentialDriveWheelSpeeds getSpeeds();
+  public DifferentialDriveWheelSpeeds getSpeeds()
+  {
+    return new DifferentialDriveWheelSpeeds(0,0);
```
Comment (cml1010101) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:108]: should return speeds. this is vital

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 4]
```diff
@@ -6,76 +6,127 @@
...
+    return 0;
+  }
   /**
    * Gets the distance traveled by the right encoder
    * @return right encoder distance in meters
    */
-  public abstract double getRightDistance();
+  public double getRightDistance()
+  {
+    return 0;
```
Comment (cml1010101) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:100]: should return right distance. This is vital

**Review by cml1010101 — CHANGES_REQUESTED** [review]

**Review by cml1010101 — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 5]
```diff
@@ -6,76 +6,130 @@
...
-
-  public abstract void driveUsingChassisSpeeds(ChassisSpeeds speeds);
-
+  public void driveVolts(double left, double right) {
+    leftSide.setVoltage(left);
+    rightSide.setVoltage(right);
+    robotDrive.feed();
+  }
+  /**
+   * Drives using feet per second
```
Comment (cml1010101) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:124]: Should be meters per second.

**Review by cml1010101 — CHANGES_REQUESTED** [review]
Just a minor change

**Review by tagnihotri — CHANGES_REQUESTED** [review]
I think it might flow better if there are some addition comments added to explain other lines of code. (Saihaj is in agreement with this statement)

**Review by cml1010101 — APPROVED** [review]

**Review by christianr-172 — APPROVED** [review]
Looks good
:)

**Review by tagnihotri — APPROVED** [review]
Good

**Pull request merged**